apps/emqx_cluster_link/src/emqx_cluster_link_api.erl

```diff
@@ -296,9 +311,9 @@ get_all_link_status_cluster() ->
                 msg => "cluster_link_api_all_status_bad_erpc_results",
                 results => BadResults
             }),
-            [];
+            ?INTERNAL_ERROR(<<"Failed to fetch cluster link status, please check the logs.">>);
```

Comment:
I'll take my suggestion back, sorry for the confusion. 🫠 I haven't pieced together the fact that it's just the resource status that gets different value if any of the RPCs have failed. Please feel free to revert that change.

My general question is basically: why log a warning if we can carry the error information with API responses, where it could be more visible and potentially more helpful? In this case, in the `node_status` field. What do you think about that?

Reply:
Good idea, I'll move the errors back to the API response.

Reply:
Remembered another reason: some terms might encode to garbled json values...

Reply:
Yes, that's what I used in the latest commits to overcome that.  But I think only recently it has started to be used, I don't spot it being used often in some APIs.

Reply:
I see, I was trying to say that it sounds to me like this is a problem to solve regardless of where error details go: in API responses or in the logs.

Reply:
I tried to push those reasons to the status and metrics API responses in the latest commits.